pints/_transformation.py

```diff
@@ -0,0 +1,1120 @@
+
+    The input parameters of the :class:`ComposedTransformation` have to be
+    ordered in the same way as the individual tranforms for the parameter
+    vector. In the above example the transform may be performed by
+    ``t.to_search(p)``, where::
+
+        p = [parameter1_transform1,
+             parameter1_transform2,
+             parameter2_transform2,
+             parameter1_transform3]
+
+    Extends :class:`Transformation`.
+    """
+    def __init__(self, *transforms):
+        # Check if sub-transforms given
+        if len(transforms) < 1:
+            raise ValueError('Must have at least one sub-transform.')
+
+        # Check if proper transform, count dimension
+        self._n_parameters = 0
+        for transform in transforms:
+            if not isinstance(transform, pints.Transformation):
+                raise ValueError('All sub-transforms must extend '
+                                 'pints.Transformation.')
+            self._n_parameters += transform.n_parameters()
+
+        # Store
+        self._transforms = transforms
+
+    def jacobian(self, q):
+        """ See :meth:`Transformation.jacobian()`. """
+        q = pints.vector(q)
+        lo, hi = 0, self._transforms[0].n_parameters()
+        output = self._transforms[0].jacobian(q[lo:hi])
+        for transform in self._transforms[1:]:
+            lo = hi
+            hi += transform.n_parameters()
+            jaco = transform.jacobian(q[lo:hi])
+            # Due to the composed transformation are independent, we can stack
+            # the Jacobian matrices J_i and J_{i+1} as
+            #
+            # J = [ J_i     0    ]
+            #     [  0   J_{i+1} ]
+            #
+            pack = np.zeros((output.shape[0], jaco.shape[1]))
+            output = np.block([[output, pack], [pack.T, jaco]])
+        return output
+
+    def jacobian_S1(self, q):
+        """ See :meth:`Transformation.jacobian_S1()`. """
+        q = pints.vector(q)
+        matrix_shape = (self.n_parameters(), self.n_parameters())
+        lo = hi = 0
+        output_S1 = np.zeros((self.n_parameters(),) + matrix_shape)
+        for transform in self._transforms:
+            lo = hi
+            hi += transform.n_parameters()
+            _, jac_S1 = transform.jacobian_S1(q[lo:hi])
+            for i, jac_S1_i in enumerate(jac_S1):
+                # Due to the composed transformation are independent, we can
+                # pack the derivative of the Jacobian matrices J_i with zeros:
+                #
+                #     [ 0   0   0 ]
+                # J = [ 0  J_i  0 ]
+                #     [ 0   0   0 ]
+                #
+                o = np.zeros(matrix_shape)
+                o[lo:hi, lo:hi] = jac_S1_i[:, :]
+                output_S1[lo + i, :, :] = o
+        return self.jacobian(q), output_S1
+
+    def multiple_to_model(self, qs):
+        """ See :meth:`Transformation.multiple_to_model()`. """
+        qs = np.asarray(qs)
```

Comment:
I don't think we can do better but I'm not loving this implementation!
You've got your original array, then a copy filled with zeros, and then copies of each part are made by the submethods!
Another reason I'd rather avoid this method 😅

Reply:
Removed this method.